rewrite-core/src/main/java/org/openrewrite/remote/RemoteArchive.java

```diff
@@ -93,7 +93,11 @@ public InputStream getInputStream(ExecutionContext ctx) {
             Path localArchive = cache.compute(uri, () -> {
                 //noinspection resource
                 HttpSender.Response response = httpSender.send(httpSender.get(uri.toString()).build());
-                return response.getBody();
+                if (response.isSuccessful()) {
```

Comment:
question to team:
should we be doing some type of retry here to improve reliability with transient errors like network glitch or delays?

Reply:
If want to implement retires, I think we might want to move this change to the `HttpSender`, I suspect other endpoints could benefit from it